packages/client/src/contexts/Primary.re

```diff
@@ -655,23 +667,26 @@ module CompetitorType = {
     | _ => Js.Exn.raiseError("Invalid Competitor Type: " ++ str)
     };
 
-  let availableInputs =
-      (~isOwner: bool, ~state: option(Types.measurableState)) => {
-    switch (isOwner, state) {
+  let availableInputs = (~measurable: Types.measurable) => {
+    let isAdmin = ChannelMembershipRole.isAdmin(measurable.channel);
+    let iAmOwner = measurable.iAmOwner |> E.Bool.O.toBool;
+    switch (iAmOwner || isAdmin, measurable.state) {
     | (true, Some(`JUDGED)) => [|`COMMENT, `OBJECTIVE, `UNRESOLVED|]
     | (true, _) => [|`COMMENT, `COMPETITIVE, `OBJECTIVE, `UNRESOLVED|]
     | (false, Some(`JUDGED)) => [|`COMMENT|]
     | (false, _) => [|`COMMENT, `COMPETITIVE|]
     };
   };
 
-  let competitorTypeInitValue =
-      (~isOwner: bool, ~state: option(Types.measurableState)) =>
-    switch (isOwner, state) {
+  let competitorTypeInitValue = (~measurable: Types.measurable) => {
+    let isAdmin = ChannelMembershipRole.isAdmin(measurable.channel);
+    let iAmOwner = measurable.iAmOwner |> E.Bool.O.toBool;
+    switch (iAmOwner || isAdmin, measurable.state) {
```

Comment:
This could be cleaned up a bit;
I suggest a function somewhere else for ``measurable.iCanEdit``

Reply:
It was one of the hardest question in my work this year. It is not a problem to add another helpful property into measurable type. It was a question how to build a subsystem of another type of security. For instance, what exactly competitor types are available for this person.

Reply:
I programming not more than 20% of my time and all another time I am trying to think about the application.

Reply:
For instance, if I am an admin of this channel what properties I can change of the entities? Here, now we should be careful. Will think about this tomorrow more.

Reply:
Makes sense. What do you mean entities? Like, Ken-knowledge graph entities, or the different aspects of the measurables?

Reply:
Makes sense, for some things it's definitely important to make good decisions, even if it takes a while.

Reply:
The entities are Measurable (question), Measurement, Agent, Bot, User, Channel, Invitation, Notification and so on. So, in a word, it is a virtual objects.